src/background.js

```diff
@@ -1,3 +1,111 @@
+// only update only the mainWindowIdOld and the mainWindowId
+function updateIcons(mainWindowId, mainWindowIdOld = null) {
+  // highlight the mainWindow
+  chrome.tabs.query({ windowId: mainWindowId }, (tabs) => {
+    tabs.forEach((tab) => {
+      // for all the mainWindow tabs
+      chrome.action.setIcon({
+        path: 'src/icon_x.png', // set the icon for only for MainWindow
+        tabId: tab.id,
+      });
+    });
+  });
+
+  // un-highlight the previous window
+  if (mainWindowIdOld !== null) {
+    chrome.tabs.query({ windowId: mainWindowIdOld }, (tabs) => {
+      tabs.forEach((tab) => {
+        // for all the mainWindow tabs
+        chrome.action.setIcon({
+          path: 'src/icon.png', // set the icon for only for MainWindow
+          tabId: tab.id,
+        });
+      });
+    });
+  }
+}
+
+// upon startup, update the icons
+chrome.storage.sync.get(
+  // run this to update the icon when the extension is installed
+  {
+    mainWindowId: 0,
+  },
+  ({ mainWindowId }) => {
+    updateIcons(mainWindowId);
+  }
+);
+
+// main icon was clicked, so toggle the mainWindow
+chrome.action.onClicked.addListener((tab) => {
+  chrome.storage.sync.get(
+    {
+      mainWindowId: 0,
+    },
+    ({ mainWindowId }) => {
+      const newWindowId =
+        mainWindowId === 0 || mainWindowId !== tab.windowId ? tab.windowId : 0;
+      updateIcons(newWindowId, mainWindowId);
+
+      chrome.storage.sync.set({
+        mainWindowId: newWindowId, // toggle it if they click the same window again
+      });
+    }
+  );
+});
+
+// it's detached, therefore this is a new window, therefore, we know it's not the mainWindow
+chrome.tabs.onDetached.addListener(async (tabId, { oldWindowId }) => {
+  chrome.storage.sync.get(
+    {
+      mainWindowId: 0,
+    },
+    ({ mainWindowId }) => {
+      if (oldWindowId === mainWindowId) {
+        chrome.action.setIcon({
+          path: 'src/icon.png', // set it as NOT the mainWindow
+          tabId,
+        });
+      }
+    }
+  );
+});
+
+// it's attached, therefore check is it needs the mainWindowIcon
+chrome.tabs.onAttached.addListener(async (tabId, { newWindowId }) => {
+  chrome.storage.sync.get(
+    {
+      mainWindowId: 0,
+    },
+    ({ mainWindowId }) => {
+      if (newWindowId === mainWindowId) {
+        chrome.action.setIcon({
+          path: 'src/icon_x.png', // set the icon for only for MainWindow
+          tabId,
+        });
+      }
+    }
+  );
+});
+
+// this is necessary because we must wait for the tab to load before setting the icon
+chrome.tabs.onUpdated.addListener(async (tabId, changeInfo, tab) => {
+  if (tab.active && changeInfo.status === 'complete') {
+    chrome.storage.sync.get(
+      {
+        mainWindowId: 0,
+      },
+      ({ mainWindowId }) => {
+        chrome.action.setIcon({
+          path:
+            tab.windowId === mainWindowId ? 'src/icon_x.png' : 'src/icon.png', // set the icon for only for MainWindow
```

Comment:
I will also think on alternatives for the icon. In my mind, "x" would mean "not", so it's counterintuitive to me that the window marked as "x" is the main window. Perhaps a green or grey dot?

Reply:
oh I agree, I was attempting to just put something up, but am fully in support of replacing it with whatever you desire.
The reason I initially put an X was, based on the original arrow, I took that to mean, "clicking the arrow makes this the mainWindow".  Therefore the X meant "clicking the X makes this NOT the main window".  But you're right, I don't think an X is the answer.  My initial take on it was I made an icon that was the window border, but filled in blue.  But looking at it, it made it hard to understand what was being shown, it just looked like a box.  I'll give the green or grey dot a shot that seems like it may be better.  The thought process is, it should convey that "this window is Active".  Thank you for the feedback!

Also thanks for explaining the "cmd-click" thing! 🎉  I WAS able to replicate that, and will patch that issue (when I have time ofc).

Also for clarity, I did not technically WANT to code things the way I did.  I could have just updated all tabs, every time a tab is focused.  However in doing so, if I have say.... 100 or more tabs open.  It can start acting laggy if it tries to update all the tabs every time, so the way it's coded right now is, it only updates the icons when it's needed.  Once it's done, I'll see if I can clean up the code at all, or abstract things a little to make it more legible.

Reply:
This is where a dropdown may help. It still is more clicks than just seeing the icon, but displaying an indicator only when you click (e.g. a checkbox in a dropdown) could simplify the logic a lot (and still provide a convenience over having to open the extension options to see which window is the main window).

Reply:
I believe I've fixed the issue with the icon not working properly if a link it middle-clicked (aka command-clicked) (previously it would cause the tab to get created, yet not be focused, which caused it to not get assigned, however now it is properly assigned).

I don't believe the menu would help as I believe it would still be global, however either way, it should be working now, but please let me know if you can find any issues with it.